File: scm-java/src/main/scheme/primitives/PrimitiveAbandonedMutexExceptionP.java
package scheme.primitives;
import scheme.*;

public class PrimitiveAbandonedMutexExceptionP extends Primitive {
    @Override
    public String name() { return "abandoned-mutex-exception?"; }

    @Override
    public String info() {
        return "Syntax: (abandoned-mutex-exception? obj)\n" +
               "Library: (srfi 18)\n" +
               "Description: Returns #t if obj is an abandoned-mutex exception.\n" +
               "Example:\n" +
               "  (abandoned-mutex-exception? exn)";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        return PrimitiveJoinTimeoutExceptionP.isThreadException(
            arguments[0], SchemeThreadException.Kind.ABANDONED_MUTEX) ? Value.T : Value.F;
    }
}

File: scm-csharp/scheme/primitives/PrimitiveAbandonedMutexExceptionP.cs
namespace scheme;

public class PrimitiveAbandonedMutexExceptionP : Primitive
{
    public override string Name() => "abandoned-mutex-exception?";

    public override string Info() =>
        "Syntax: (abandoned-mutex-exception? obj)\n" +
        "Library: (srfi 18)\n" +
        "Description: Returns #t if obj is an abandoned-mutex exception.\n" +
        "Example:\n" +
        "  (abandoned-mutex-exception? exn)";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        return PrimitiveJoinTimeoutExceptionP.IsThreadException(
            arguments[0], ThreadExceptionKind.ABANDONED_MUTEX) ? Value.T : Value.F;
    }
}
